lib/dataloader/ecto.ex

```diff
@@ -894,10 +894,15 @@ if Code.ensure_loaded?(Ecto) do
       end
 
       defp maybe_distinct(query, [%Ecto.Association.Has{}, %Ecto.Association.BelongsTo{} | _]) do
-        distinct(query, true)
+        %Ecto.Query{distinct: dist} = query
```

Comment:
Would this be cleaner if it was pattern-matched in the function arguments?
For example
```
defp maybe_distinct(%Ecto.Query{distinct: dist} = query, _) when dist, do: query
```
Then the rest of the patterns can assume `dist` is false or nil.
```
defp maybe_distinct(query, [%Ecto.Association.Has{}, %Ecto.Association.BelongsTo{} | _]), do: distinct(query, true)

defp maybe_distinct(query, [%Ecto.Association.ManyToMany{} | _]), do: distinct(query, true)
```